src/test/mochitest/browser_dbg-sourcemapped-scopes.js

```diff
@@ -28,513 +28,1134 @@ add_task(async function() {
+    ]);
+  }
+}
 
-      // This value is currently optimized away, which isn't 100% accurate.
-      // Because import declarations is the last thing in the file, our current
-      // logic doesn't cover _both_ 'var' statements that it generates,
-      // making us use the first, optimized-out binding. Given that imports
-      // are almost never the last thing in a file though, this is probably not
-      // a huge deal for now.
-      ["aDefault", "(optimized away)"],
-      ["root", "(optimized away)"],
-      ["val", "(optimized away)"]
-    ]
-  );
-
+async function testWebpackLineMappings(dbg) {
   await breakpointScopes(
     dbg,
-    "webpack3-babel6",
-    "babel-flowtype-bindings",
-    { line: 9, column: 2 },
+    "webpack3",
+    "webpack-line-mappings",
+    { line: 11, column: 0 },
     [
-      "Module",
-      ["aConst", '"a-const"'],
-      ["Four", "Getter"],
+      "Block",
+      ["<this>", '"this-value"'],
+      ["arg", '"arg-value"'],
+      ["arguments", "Arguments"],
+      ["inner", "undefined"],
+      "Block",
+      ["someName", "(optimized away)"],
+      "Block",
+      ["two", "2"],
+      "Block",
+      ["one", "1"],
+      "root",
+      ["arguments", "Arguments"],
+      "fn:someName()",
+      "webpack3WebpackLineMappings",
+      ["__webpack_exports__", "(optimized away)"],
+      ["__WEBPACK_IMPORTED_MODULE_0__src_mod1__", "{\u2026}"],
+      ["__webpack_require__", "(optimized away)"],
+      ["arguments", "(unavailable)"],
+      ["module", "(optimized away)"],
       "root()"
     ]
   );
+}
 
-  await breakpointScopes(dbg, "webpack3-babel6", "babel-switches", { line: 7, column: 6 }, [
-    "Switch",
-    ["val", "2"],
-    "Function Body",
-    ["val", "1"],
-    "Module",
-    "root()"
-  ]);
-
-  await breakpointScopes(dbg, "webpack3-babel6", "babel-switches", { line: 10, column: 6 }, [
+async function testWebpackFunctions(dbg) {
+  await breakpointScopes(dbg, "webpack3", "webpack-functions", { line: 4, column: 0 }, [
     "Block",
-    ["val", "3"],
-    "Switch",
-    ["val", "2"],
-    "Function Body",
-    ["val", "1"],
-    "Module",
-    "root()"
+    ["<this>", "{\u2026}"],
+    ["arguments", "Arguments"],
+    ["x", "4"],
+    "webpack3WebpackFunctions",
+    ["__webpack_exports__", "(optimized away)"],
+    ["__webpack_require__", "(optimized away)"],
+    ["arguments", "(unavailable)"],
+    ["module", "{\u2026}"],
+    ["root", "(optimized away)"]
   ]);
+}
 
-  await breakpointScopes(dbg, "webpack3-babel6", "babel-try-catches", { line: 8, column: 4 }, [
-    "Block",
-    ["two", "2"],
-    "Catch",
-    ["err", '"AnError"'],
-    "Function Body",
-    ["one", "1"],
-    "Module",
-    "root()"
-  ]);
+async function testESModules(dbg) {
+  // TODO: The behavior on Webpack 3 seems to be broken.
+  await breakpointScopes(
+    dbg,
+    "webpack3",
+    "esmodules",
+    { line: 20, column: 0 },
+    [
+      "Module",
+      ["aDefault", "(optimized away)"],
```

Comment:
FYI, one benefit of expanding these is that I came across this. Definitely not desirable. This should be falling back to generated bindings, but isn't. If there were more non-import bindings, it'd tip it over the edge I think, but since this file is almost exclusively import bindings and they are all detected as optimized out, it isn't falling back properly.

Reply:
Ooh, nice find

Reply:
Fixed by #6710